PriceKataCalc/Program.cs

```diff
@@ -59,27 +68,56 @@ public double calcDiscount()
             double myprice;
             if (this.discount.beforeTax)
             {
-            myprice= this.price *this.discount.discountValue;
+                myprice = this.price * this.discount.discountValue;
+                myprice = Limit(myprice);
             }
             else
             {
-            myprice= calcPriceAfterTax() *this.discount.discountValue;
+                myprice = calcPriceAfterTax() * this.discount.discountValue;
+                myprice = Limit(myprice);
+
             }
             return myprice;
         }
+
+        private double Limit(double myprice)
+        {
+            if (!Discount.CAP.isPercentage)
```

Comment:
Same here, why should the product know about the cap if it is percentage or flat cost ?